File: scripts/take_screenshots.py
"""Capture README screenshots for each page."""
import asyncio
from pathlib import Path
from playwright.async_api import async_playwright

BASE = "http://localhost:5173"
OUT = Path("/Users/kanuj/regime-radar/docs/screenshots")
OUT.mkdir(parents=True, exist_ok=True)

PAGES = [
    ("current-state",    "/",                None),
    ("history",          "/history",         None),
    ("event-replay",     "/event-replay",    None),
    ("model-drivers",    "/model-drivers",   None),
    ("scenario-explorer", "/scenario",       "crisis_peak"),
]

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        ctx = await browser.new_context(
            viewport={"width": 1440, "height": 860},
            device_scale_factor=2,
        )
        page = await ctx.new_page()
        # Hide scrollbars globally
        await page.add_style_tag(content="""
            ::-webkit-scrollbar { display: none !important; }
            * { scrollbar-width: none !important; }
        """)

        for slug, path, preset in PAGES:
            print(f"  {slug}...")
            await page.goto(f"{BASE}{path}", wait_until="networkidle")
            # Wait for any loading states to resolve
            await page.wait_for_timeout(2500)

            # For scenario explorer, click the Crisis Peak preset then wait
            if preset == "crisis_peak":
                try:
                    await page.click("text=Crisis Peak", timeout=4000)
                    await page.wait_for_timeout(2500)
                except Exception:
                    pass

            # For event replay, click the COVID 2020 event if available
            if slug == "event-replay":
                try:
                    await page.click("text=COVID", timeout=3000)
                    await page.wait_for_timeout(1500)
                except Exception:
                    pass

            await page.screenshot(
                path=str(OUT / f"{slug}.png"),
                full_page=False,
                clip={"x": 0, "y": 0, "width": 1440, "height": 860},
            )
            print(f"  ✓ {slug}.png saved")

        await browser.close()
    print("Done.")

asyncio.run(main())

File: scripts/take-screenshots.js
// Screenshot script — run with: node scripts/take-screenshots.js
const { chromium } = require('/Users/kanuj/regime-radar/frontend/node_modules/@playwright/test')
const path = require('path')

const BASE = 'http://localhost:5175'
const OUT = path.join(__dirname, '..', 'docs', 'screenshots')

async function waitForNoSpinner(page) {
  // Give animations and data fetches time to settle
  await page.waitForLoadState('networkidle')
  await page.waitForTimeout(1200)
}

;(async () => {
  const browser = await chromium.launch({ channel: 'chrome', headless: true })
  const ctx = await browser.newContext({
    viewport: { width: 1440, height: 900 },
    deviceScaleFactor: 2,
  })
  const page = await ctx.newPage()

  // ── Current State ──────────────────────────────────────────────────────────
  console.log('📸 current-state...')
  await page.goto(`${BASE}/`, { waitUntil: 'networkidle' })
  await waitForNoSpinner(page)
  await page.screenshot({ path: `${OUT}/current-state.png`, fullPage: false })

  // ── History ────────────────────────────────────────────────────────────────
  console.log('📸 history...')
  await page.goto(`${BASE}/history`, { waitUntil: 'networkidle' })
  await waitForNoSpinner(page)
  await page.screenshot({ path: `${OUT}/history.png`, fullPage: false })

  // ── Event Replay ───────────────────────────────────────────────────────────
  console.log('📸 event-replay...')
  await page.goto(`${BASE}/event-replay`, { waitUntil: 'networkidle' })
  await waitForNoSpinner(page)
  await page.screenshot({ path: `${OUT}/event-replay.png`, fullPage: false })

  // ── Signal Breakdown (Model Drivers) ──────────────────────────────────────
  console.log('📸 signal-breakdown (model-drivers)...')
  await page.goto(`${BASE}/model-drivers`, { waitUntil: 'networkidle' })
  await waitForNoSpinner(page)
  await page.screenshot({ path: `${OUT}/model-drivers.png`, fullPage: false })

  // ── Scenario Explorer — Crisis Peak ────────────────────────────────────────
  console.log('📸 scenario-explorer (Crisis Peak)...')
  await page.goto(`${BASE}/scenario`, { waitUntil: 'networkidle' })
  await waitForNoSpinner(page)

  // Click Crisis Peak preset
  const crisisPeak = page.locator('text=Crisis Peak')
  if (await crisisPeak.isVisible()) {
    await crisisPeak.click()
    await page.waitForTimeout(1500) // let driver cards animate in
  } else {
    console.warn('  ⚠ Crisis Peak button not found — falling back to default state')
  }

  await page.screenshot({ path: `${OUT}/scenario-explorer.png`, fullPage: false })

  await browser.close()
  console.log('✅ All screenshots saved to docs/screenshots/')
})()
